geode-core/src/main/java/org/apache/geode/internal/cache/execute/PartitionedRegionFunctionResultSender.java

```diff
@@ -81,31 +86,48 @@ public KnownVersion getClientVersion() {
     return null;
   }
 
+  public PartitionedRegionFunctionResultSender(DistributionManager dm, PartitionedRegion pr,
+      long time, PartitionedRegionFunctionStreamingMessage msg,
+      Function function, int[] bucketArray) {
+    this(dm, pr, time, msg, function, bucketArray,
+        (x, y) -> FunctionStatsManager.getFunctionStats((String) x, (InternalDistributedSystem) y));
+  }
+
   /**
    * Have to combine next two constructor in one and make a new class which will send Results back.
    *
    */
   public PartitionedRegionFunctionResultSender(DistributionManager dm, PartitionedRegion pr,
```

Comment:
If this new constructor is used only testing it should be package private. Evaluate access level on each of these constructors.